Modules/timemodule.c

```diff
@@ -2390,3 +2382,67 @@ pysleep(PyTime_t timeout)
     return -1;
 #endif
 }
+
+
+// time.sleep(0) optimized implementation.
+// On error, raise an exception and return -1.
+// On success, return 0.
+//
+// Rationale
+// ---------
+// time.sleep(0) is slower when using the generic implementation, but we make
+// it faster than time.sleep(eps) for eps > 0 so to avoid some performance
+// annoyance. For details, see https://github.com/python/cpython/pull/128274.
+static int
+pysleep_zero(void)
+{
+    assert(!PyErr_Occurred());
+#ifndef MS_WINDOWS
+    int ret, err;
+    Py_BEGIN_ALLOW_THREADS
+#ifdef HAVE_CLOCK_NANOSLEEP
+    struct timespec zero = {0, 0};
+    ret = clock_nanosleep(CLOCK_MONOTONIC, TIMER_ABSTIME, &zero, NULL);
```

Comment:
I don't think that TIMER_ABSTIME is appropriate here:

```suggestion
    ret = clock_nanosleep(CLOCK_MONOTONIC, 0, &zero, NULL);
```

Reply:
I first thought about:

> If flags is TIMER_ABSTIME, then t is interpreted as an absolute
  time as measured by the clock, clockid. If t is less than or
  equal to the current value of the clock, then clock_nanosleep()
  returns immediately without suspending the calling thread.

Without this flag, `time.sleep(0)` takes 50us. Otherwise it takes 2us. So, now I'm more and more inclined to actually revert it back to a `select`. Because otherwise, it's as if I'm not doing anything (and just skip the call to `clock_nanosleep`; I mean I already know that the condition on `t` is verified so it's as if I have no

Reply:
I'm leaving for a few days but I'm still struggling to convince myself between the use of `select` or not. The reason why `clock_nanosleep()` is slowed down is because we don't pass a zero time struct.

Reply:
Ok, I was wrong. See https://github.com/python/cpython/pull/128274#issuecomment-2575513339.